File: nutrition/src/main/java/com/example/nutrition/dao/UserDao.java
package com.example.nutrition.dao;

import org.apache.ibatis.annotations.Insert;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Select;
import org.apache.ibatis.annotations.Update;

import com.example.nutrition.entity.User;

@Mapper
public interface UserDao {
    @Insert("insert into users (username, password) values (#{username}, #{password})")
    public Integer insertUser(String username, String password);

    @Select("select password from users where username=#{username}")
    public String getPassword(String username);

    @Select("select * from users where username=#{username}")
    public Integer isRegistered(String username);

    @Update("update users set username=#{newUsername} where username=#{oldUsername}")
    public Integer updateUsername(String oldUsername, String newUsername);

    @Update("update users set password=#{newPassword} where username=#{username}")
    public Integer updatePassword(String username, String newPassword);

    @Update("update users set sex=#{sex}, age=#{age}, job=#{job}, signature=#{signature}, region=#{region} where username=#{username}")
    public Integer updateDetails(User user);

    @Update("update users set region=#{region} where id=#{user_id}")
    public Integer updateRegion(int user_id, String region);

    @Update("update users set job=#{job} where id=#{user_id}")
    public Integer updateJob(int user_id, String job);

    @Select("select * from users where username=#{username}")
    public User getByUsername(String username);

    @Select("select name from food where id=#{id}")
    public String getFoodNameById(int id);

    @Select("select id from food where name like #{name} limit 1")
    public Integer getLikelyFoodIdByName(String name);
}

File: spitts/src/dao/UserDao.go
package dao

import (
	"spitts/src/entity"
	"strconv"
)

func IsRegistered(username string) bool {
	var count int
	query := "select count(*) from users where username = $1"
	entity.Database.DB.QueryRow(query, username).Scan(&count)

	return count != 0
}

func CheckPassword(username string, password string) bool {
	var count int
	query := "select count(*) from users where username = $1 and password = $2"
	entity.Database.DB.QueryRow(query, username, password).Scan(&count)

	return count != 0
}

func SelectUser(username string) entity.User {
	var user entity.User
	query := "select * from users where username = $1"
	entity.Database.DB.QueryRow(query, username).Scan(&user.Username, &user.Password, &user.Type)

	return user
}

func InsertUser(username string, password string, user_type int) error {
	user_t := strconv.Itoa(user_type)
	query := "insert into users(username, password, type) values($1, $2, $3)"
	_, err := entity.Database.DB.Exec(query, username, password, user_t)

	return err
}

func UpdateUsername(old_username string, new_username string) error {
	query := "update users set username = $1 where username = $2"
	_, err := entity.Database.DB.Exec(query, new_username, old_username)

	return err
}

func UpdatePassword(username string, password string) error {
	query := "update users set password = $1 where username = $2"
	_, err := entity.Database.DB.Exec(query, password, username)

	return err
}
